### PR TITLE
Run doc generation and test using non-dev version

### PR DESCRIPTION
**Pull request by praveenkuttappan**
Currently scheduled nightly CI runs doc generation and test on dev build version. This is causing an issue when dependent package version is not yet available on PyPI and package needs future GA version or higher as per install requires. for e.g. scheduled CI for eventhub is failing now since version is changed to 5.0.0 and it is not yet released. dev version of this package 5.0.0.devxxxxxxxxxxx is lower than 5.0.0. eventhub-checkpointstore has a requirement of >= 5.0.0 which will not be satisfied if version is updated to dev in the repo.

Solution: Update the version temporarily only when building dev build package. Test and doc gen should run on non-dev version.

### CODE REVIEW

File: eng/pipelines/templates/steps/build-artifacts.yml [hunk 1]
```diff
@@ -64,6 +59,18 @@ steps:
...
         --toxenv=sphinx
 
+  - template: ../steps/set-dev-build.yml
+    parameters:
+      ServiceDirectory: ${{ parameters.ServiceDirectory }}
+      BuildTargetingString: ${{ parameters.BuildTargetingString }}
+
+  - task: PythonScript@0
+    displayName: 'Generate Dev Build Packages'
+    condition: eq(variables['SetDevVersion'],'true')
```
Comment (weshaggard) [inline eng/pipelines/templates/steps/build-artifacts.yml]: Isn't this building the packages twice? How do we ensure they we don't have both sets published to artifacts?

**Comment by weshaggard** [comment]
I'm not sure I fully understand this change. I thought we wanted to test against the dev versions to ensure we are validating the dev packages we are publishing to nightly?

**Comment by weshaggard** [comment]
> scheduled CI for eventhub is failing now since version is changed to 5.0.0 and it is not yet released. dev version of this package 5.0.0.devxxxxxxxxxxx is lower than 5.0.0. eventhub-checkpointstore has a requirement of >= 5.0.0 which will not be satisfied if version is updated to dev in the repo.

In that example it seems like the missing piece is that the dependency needs to allow for dev/prerelease versions of the 5.0.0 package. It is being too strict in its dependency.

**Comment by praveenkuttappan** [comment]
Not required for now

**Pull request closed without merging**